src/topp/OpenSwathFeatureXMLToTSV.cpp

```diff
@@ -291,19 +290,17 @@ void write_out_body_(std::ostream &os, Feature *feature_it, TargetedExperiment &
   }
   else
   {
-    char mz_char[40];
     for (std::vector<Feature>::iterator sub_it = feature_it->getSubordinates().begin(); sub_it != feature_it->getSubordinates().end(); ++sub_it)
     {
       os.precision(writtenDigits(double()));
-      sprintf(intensity_char, "%f", sub_it->getIntensity());
-      sprintf(mz_char, "%f", sub_it->getMZ());
+      String intensity_area = boost::str(boost::format("%f") % sub_it->getIntensity());
+      String mz_str = boost::str(boost::format("%f") % sub_it->getMZ());
       String apex = "NA";
       if (sub_it->metaValueExists("peak_apex_int"))
       {
-        sprintf(intensity_apex_char, "%f", (double)sub_it->getMetaValue("peak_apex_int"));
-        apex = (String) intensity_apex_char;
+        apex = boost::str(boost::format("%f") % sub_it->getMetaValue("peak_apex_int"));
       }
-      os << line << meta_values << (String)intensity_char << "\t" << apex << "\t" << (String)sub_it->getMetaValue("native_id") << "\t" << (String)mz_char << std::endl;
+      os << line << meta_values << intensity_area << "\t" << apex << "\t" << (String)sub_it->getMetaValue("native_id") << "\t" << mz_str << std::endl;
```

Comment:
maybei its sufficient to use
os.imbue(std::locale("En_US"));
and directly write out floats / ints